File: 57-insert-interval/57-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        
        vector<vector<int>> ans;
        
        
        vector<int> temp = newInterval;
        
        for(auto it:intervals){
            
            if(it[0]<=temp[1] and it[1]>=temp[0]){
                temp[0] = min(it[0],temp[0]);
                temp[1] = max(it[1],temp[1]);
                
            } else if(temp[1]<it[0]){
              ans.push_back(temp);
               temp = it; 
            } else{
                ans.push_back(it);
                
            }
            
        }
        
        ans.push_back(temp);
        
        
        return ans;
        
        
        
        
        
    }
};

File: 57-insert-interval/57-insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        
        ans = []
        
        for i in range(len(intervals)):
            
            # less than current
            if newInterval[1]<intervals[i][0]:
                ans.append(newInterval)
                return ans + intervals[i:]
                            
            elif newInterval[0] > intervals[i][1]:  ## if bigger than current
                ans.append(intervals[i])
            
            else:
                
                # if overlapping on either side
                
                # update min and max of newinterval
                newInterval = [min(newInterval[0],intervals[i][0]),
                               max(newInterval[1],intervals[i][1])]
            
        ans.append(newInterval) # if it is not returned in 1st case then in end we need to append
        
        return ans
            
            
            
        